### PR TITLE
Create ForgeRock_CEF.json

### PR DESCRIPTION
**Pull request by javaservlets**
Fixes #
First PR for the ForgeRock CEF connector
## Proposed Changes

  -
  -
  -

### CODE REVIEW

File: DataConnectors/ForgeRock_CEF.json [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+          "write" : true,
+          "delete" : true
+        }
+      }
+    ]
+  },
+  "instructionSteps" : [
+    {
+      "title" : "Configuration for the ForgeRock Common Audit (CAUD) for Microsoft Azure Sentinel",
+      "description" : "Install and configure the ForgeRock Common Audit (CAUD) for Microsoft Azure Sentinel per the documentation at https://github.com/javaservlets/SentinelAuditEventHandler",
```
Comment (preetikr) [inline DataConnectors/ForgeRock_CEF.json:58]: I reviewed the documentation hosted on GitHub https://github.com/javaservlets/SentinelAuditEventHandler - please update all references of "Sentinel" to "Azure Sentinel" which is the brand name of the product in public documentation.

File: DataConnectors/ForgeRock_CEF.json [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  "publisher" : "ForgeRock Inc",
+  "descriptionMarkdown" : "The ForgeRock Identity Platform provides a single common auditing framework. Extract and aggregate log data across the entire platform with common audit (CAUD) event handlers and unique IDs so that it can be tracked holistically. Open and extensible, you can leverage audit logging and reporting capabilities for integration with Microsoft Azure Sentinel via this CAUD for CEF connector.",
+  "graphQueries" : [
+    {
+      "metricName" : "Total data received",
+      "legend" : "DATATYPE_NAME",
+      "baseQuery" : "CommonSecurityLog​ | where DeviceVendor in (\"ForgeRock Inc\")\n| where DeviceAction =~ \"FAILED\""
+    }
+  ],
+  "sampleQueries" : [
```
[the remark this answers is not in the document]
Comment (javaservlets) [reply DataConnectors/ForgeRock_CEF.json:13]: I've added DeviceVendor per your instructions; 'Appliance' is not really applicable to our connector but we can explore that in a future version

File: DataConnectors/ForgeRock_CEF.json [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    {
+      "name" : "CommonSecurityLog (DATATYPE_NAME)",
+      "lastDataReceivedQuery" : "CommonSecurityLog | where DeviceVendor in (\"ForgeRock Inc\")\n| where DeviceProduct == \"Trust Partner Network\"\n| summarize Time = max(TimeGenerated)\n| where isnotempty(Time)"
+    }
+  ],
+  "connectivityCriterias" : [
+    {
+      "type" : "IsConnectedQuery",
+      "value" : [
+        "\nCommonSecurityLog\n| where DeviceVendor == \"PROVIDER NAME\"\n| where DeviceProduct == \"APPLIANCE NAME\"\n\n            | summarize LastLogReceived = max(TimeGenerated)\n            | project IsConnected = LastLogReceived > ago(30d)"
```
[the remark this answers is not in the document]
Comment (javaservlets) [reply DataConnectors/ForgeRock_CEF.json:33]: 'Appliance' is not really applicable to our connector but we can explore that in a future version

File: DataConnectors/ForgeRock_CEF.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (javaservlets) [reply DataConnectors/ForgeRock_CEF.json:13]: Hi... I believe I've just committed the updates to this json file as instructed.

File: DataConnectors/ForgeRock_CEF.json (hunk 1 above)
Comment (javaservlets) [reply DataConnectors/ForgeRock_CEF.json]: done

File: DataConnectors/ForgeRock_CEF.json [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+      "query" : "CommonSecurityLog | where DeviceVendor in (\"ForgeRock Inc\")\n| extend user = split(split(AdditionalExtensions, \";\", 4)[0], \"=\", 1)[0] // extract userID from  4th field in json payload\n| where strlen(user) > 2 // if not null\n| where DeviceAction =~ \"FAILED\""
+    },
+    {
+      "description" : "Autocluster logins by IP addresses ",
+      "query" : "CommonSecurityLog | where DeviceVendor in (\"ForgeRock Inc\")\n| where DeviceAction =~ \"FAILED\"\n| project SourceIP \n| evaluate autocluster(0.6)"
+    }
+  ],
+  "dataTypes" : [
+    {
+      "name" : "CommonSecurityLog (DATATYPE_NAME)",
```
[the remark this answers is not in the document]
Comment (javaservlets) [reply DataConnectors/ForgeRock_CEF.json]: done

File: DataConnectors/ForgeRock_CEF.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (preetikr) [reply DataConnectors/ForgeRock_CEF.json:13]: In the base query please remove DeviceAction construct. This is a total data query so includes all types of actions.

File: DataConnectors/ForgeRock_CEF.json [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+    {
+      "metricName" : "Total data received",
+      "legend" : "ForgeRock",
+      "baseQuery" : "CommonSecurityLog | where DeviceVendor in (\"ForgeRock Inc\") | where DeviceProduct in (\"Trust Partner Network\")\n| where DeviceAction =~ \"FAILED\""
+    }
+  ],
+  "sampleQueries" : [
+    {
+      "description" : "Total failed login attempts to the ForgeRock Platform",
+      "query" : "CommonSecurityLog | where DeviceVendor in (\"ForgeRock Inc\") | where DeviceProduct in (\"Trust Partner Network\")\n| extend user = split(split(AdditionalExtensions, \";\", 4)[0], \"=\", 1)[0] // extract userID from  4th field in json payload\n| where strlen(user) > 2 // if not null\n| where DeviceAction =~ \"FAILED\""
```
Comment (preetikr) [inline DataConnectors/ForgeRock_CEF.json:16]: The sample queries are pretty complicated. The goal is not only to get data into Azure Sentinel but also in a format that enables customers to easily work with the data in different Azure Sentinel experiences. I have provided feedback on the sample data as well. Please map the values embedded in msg and other fields into relevant CEF columns. That way the queries become very simple. The queries are ultimately used in building workbooks, hunting queries, analytics etc. so having a simple way to work with the data in columnar formats is very critical.

File: DataConnectors/ForgeRock_CEF.json [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+    {
+      "name" : "CommonSecurityLog (ForgeRock)",
+      "lastDataReceivedQuery" : "CommonSecurityLog | where DeviceVendor in (\"ForgeRock Inc\") | where DeviceProduct == \"Trust Partner Network\"\n| summarize Time = max(TimeGenerated)\n| where isnotempty(Time)"
+    }
+  ],
+  "connectivityCriterias" : [
+    {
+      "type" : "IsConnectedQuery",
+      "value" : [
+        "\nCommonSecurityLog\n| where DeviceVendor == \"PROVIDER NAME\"\n| where DeviceProduct == \"APPLIANCE NAME\"\n\n            | summarize LastLogReceived = max(TimeGenerated)\n            | project IsConnected = LastLogReceived > ago(30d)"
```
Comment (preetikr) [inline DataConnectors/ForgeRock_CEF.json:33]: As you've already done with other queries above, replace "PROVIDER_NAME" with "ForgeRock Inc" and DeviceProduct with "Trust Partner Network". You already have the device provider and devoce vendor (name of your product) in all queries. Extend the same to the connectivityCriterias query as well.

File: DataConnectors/ForgeRock_CEF.json [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+      "title" : "\n\n\n1. Linux Syslog agent configuration",
+      "description" : "Install and configure the Linux agent to collect your Common Event Format (CEF) Syslog messages and forward them to Azure Sentinel.\n\n> Notice that the data from all regions will be stored in the selected workspace",
+      "innerSteps" : [
+        {
+          "title" : "1.1 Select or create a Linux machine",
+          "description" : "Select or create a Linux machine that Azure Sentinel will use as the proxy between your security solution and Azure Sentinel this machine can be on your on-prem environment, Azure or other clouds."
+        },
+        {
+          "title" : "1.2 Install the CEF collector on the Linux machine",
+          "description" : "Install the Microsoft Monitoring Agent on your Linux machine and configure the machine to listen on the necessary port and forward messages to your Azure Sentinel workspace. The CEF collector colects CEF messages on port 514 TCP.\n\n> 1. Make sure that you have Python on your machine using the following command: python –version.\n\n> 2. You must have elevated permissions (sudo) on your machine.",
```
Comment (preetikr) [inline DataConnectors/ForgeRock_CEF.json:70]: Please fix the typo - collects (it currently shows as colects).

File: DataConnectors/ForgeRock_CEF.json [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+                "value" : "sudo wget https://raw.githubusercontent.com/Azure/Azure-Sentinel/master/DataConnectors/CEF/cef_installer.py&&sudo python cef_installer.py {0} {1}"
+              },
+              "type" : "CopyableLabel"
+            }
+          ]
+        }
+      ]
+    },
+    {
+      "title" : "2. Forward Common Event Format (CEF) logs to Syslog agent",
```
Comment (preetikr) [inline DataConnectors/ForgeRock_CEF.json:88]: I see weird characters in step 2 after "port 514 TCP on the machineâs IP address." Please delete the text "machine's IP" and retype to ensure it renders fine. You should be able to validate and see if this is fixed by the test utility https://portal.azure.com/?feature.BringYourOwnConnector=true per the data connector guidance. Please test and fix these.

File: DataConnectors/ForgeRock_CEF.json [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+      "title" : "Configuration for the ForgeRock Common Audit (CAUD) for Microsoft Azure Sentinel",
+      "description" : "Install and configure the ForgeRock Common Audit (CAUD) for Microsoft Azure Sentinel per the documentation at https://github.com/javaservlets/SentinelAuditEventHandler",
+      "innerSteps" : [
+        {
+          "title" : "1.1 Select or create a Linux machine",
+          "description" : "Select or create a Linux machine that Azure Sentinel will use as the proxy between your security solution and Azure Sentinel this machine can be on your on-prem environment, Azure or other clouds."
+        },
+        {
+          "title" : "1.2 Install the CEF collector on the Linux machine",
+          "description" : "Install the Microsoft Monitoring Agent on your Linux machine and configure the machine to listen on the necessary port and forward messages to your Azure Sentinel workspace. The CEF collector colects CEF messages on port 514 TCP.\n\n> 1. Make sure that you have Python on your machine using the following command: python –version.\n\n> 2. You must have elevated permissions (sudo) on your machine.",
```
[the remark this answers is not in the document]
Comment (preetikr) [reply DataConnectors/ForgeRock_CEF.json:70]: This comment is still unresolved. Please remove the weird characters after "python âversion". You should be able to validate and see if this is fixed by the test utility https://portal.azure.com/?feature.BringYourOwnConnector=true per the data connector guidance. Please test and fix these.

File: DataConnectors/ForgeRock_CEF.json [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+        }
+      ]
+    },
+    {
+      "title" : "2. Forward Common Event Format (CEF) logs to Syslog agent",
+      "description" : "Set your security solution to send Syslog messages in CEF format to the proxy machine. Make sure you to send the logs to port 514 TCP on the machine’s IP address."
+    },
+    {
+      "title" : "3. Validate connection",
+      "description" : "Follow the instructions to validate your connectivity:\n\nOpen Log Analytics to check if the logs are received using the CommonSecurityLog schema.\n\n>It may take about 20 minutes until the connection streams data to your workspace.\n\nIf the logs are not received, run the following connectivity validation script:\n\n> 1. Make sure that you have Python on your machine using the following command: python –version\n\n>2. You must have elevated permissions (sudo) on your machine",
```
Comment (preetikr) [inline DataConnectors/ForgeRock_CEF.json:93]: Please remove the weird characters after "python âversion" and replace with hyphen. You should be able to validate and see if this is fixed by the test utility https://portal.azure.com/?feature.BringYourOwnConnector=true per the data connector guidance. Please test and fix these.

File: DataConnectors/ForgeRock_CEF.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (javaservlets) [reply DataConnectors/ForgeRock_CEF.json:13]: I've removed it from basequery but I don't see how to test it from https://portal.azure.com/?feature.BringYourOwnConnector

File: DataConnectors/ForgeRock_CEF.json (hunk 6 above)
Comment (javaservlets) [reply DataConnectors/ForgeRock_CEF.json:33]: done

File: DataConnectors/ForgeRock_CEF.json (hunk 8 above)
Comment (javaservlets) [reply DataConnectors/ForgeRock_CEF.json:88]: I don't see those characters... could be a MacOS thing?

Comment (javaservlets) [reply DataConnectors/ForgeRock_CEF.json:88]: Ie, I always modify the json, upload to ?feature.BringYourOwnConnector first, verify sencond, and only then copy and paste into github. somewhere in that last step things are weird; and note those are the OOTB microsoft instructions I am copying and pasting

File: DataConnectors/ForgeRock_CEF.json (hunk 9 above)
[the remark this answers is not in the document]
Comment (javaservlets) [reply DataConnectors/ForgeRock_CEF.json]: done

File: DataConnectors/ForgeRock_CEF.json (hunk 7 above)
Comment (javaservlets) [reply DataConnectors/ForgeRock_CEF.json]: done

File: DataConnectors/ForgeRock_CEF.json (hunk 8 above)
Comment (preetikr) [reply DataConnectors/ForgeRock_CEF.json:88]: I can still see the ASCII chars in both places. This PR has been in progress for a long time. I'll approve and I'll make the make the change

**Pull request merged**